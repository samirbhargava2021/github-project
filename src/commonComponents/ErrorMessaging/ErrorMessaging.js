import React from "react";
import "./ErrorMessaging.scss";

const ErrorMessaging = ({ text }) => {
  return (
    <>
    {!!text&&
      <p className="error-position">{text}</p>
    }
    </>
  );
};

export default ErrorMessaging;
