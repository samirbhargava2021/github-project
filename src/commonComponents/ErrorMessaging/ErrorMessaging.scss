.error-position {
  display: flex;
  justify-content: center;
  color: #e91212;
}
