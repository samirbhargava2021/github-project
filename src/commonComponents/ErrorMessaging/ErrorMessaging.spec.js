import { shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
import ErrorMessaging from "./ErrorMessaging";

describe("Error Component", () => {
  it("should render Error Component", () => {
    const component = shallow(<ErrorMessaging error="0 results" />);
    expect(component.getElements()).toMatchSnapshot();
  });
  it("should check if null is passed in text and element shoud not render", () => {
    render(<ErrorMessaging text="" />);
    const buttontxt = screen.queryByText('0 results');
    expect(buttontxt).toBeNull();
  });
  it("should have text passed as props to the component", () => {
    const { getByText } = render(<ErrorMessaging text="0 results" />);
    expect(getByText("0 results")).toBeDefined();
  });
});
