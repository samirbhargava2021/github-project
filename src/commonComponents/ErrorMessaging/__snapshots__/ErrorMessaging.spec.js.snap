// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render Error Component 1`] = `
Array [
  <React.Fragment />,
]
`;
